src/gentropy/method/ld_matrix_interface.py

```diff
@@ -0,0 +1,104 @@
+"""Step to import filtered version of a LD matrix (block matrix)."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pyspark.sql.functions as f
+
+from gentropy.common.session import Session
+from gentropy.datasource.gnomad.ld import GnomADLDMatrix
+from gentropy.datasource.pan_ukbb_ld.ld import PanUKBBLDMatrix
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame, Row
+
+
+class LDMatrixInterface:
+    """Toolset to interact with LD matrices."""
+
+    ancestry_map = {
+        "nfe": "EUR",
+        "csa": "CSA",
+        "afr": "AFR",
+    }
+
+    @staticmethod
+    def get_locus_index_boundaries(
+        session: Session,
+        study_locus_row: Row,
+        ancestry: str = "nfe",
+    ) -> DataFrame:
+        """Extract hail matrix index from StudyLocus rows.
+
+        Args:
+            session (Session): Session object
+            study_locus_row (Row): Study-locus row
+            ancestry (str): Major population to extract from gnomad matrix, default is "nfe"
+
+        Returns:
+            DataFrame: Returns the index of the gnomad matrix for the locus
+
+        """
+        mapped_ancestry = LDMatrixInterface.ancestry_map.get(ancestry, ancestry)
+
+        if (ancestry in ("nfe", "csa", "afr")) and (session is not None):
+            joined_index = PanUKBBLDMatrix().get_locus_index_boundaries(
+                session=session,
+                study_locus_row=study_locus_row,
+                ancestry=mapped_ancestry,
```

Comment:
```suggestion
                ancestry=LDMatrixInterface.ancestry_map.get(ancestry),
```
At this point we already know that these ancestries are directed to the PanUKBB path.